blog/docs/articles/workflow-as-function-flow.md

```diff
@@ -0,0 +1,328 @@
+gcloud eventarc triggers create generateusernameandpassword --event-filters="type=google.cloud.pubsub.topic.v1.messagePublished" --destination-run-service=user-registration-gcp-cloudrun --destination-run-path=/ --transport-topic=io.automatiko.examples.userRegistration.generateusernameandpassword --location=us-central1
+gcloud eventarc triggers create getuser --event-filters="type=google.cloud.pubsub.topic.v1.messagePublished" --destination-run-service=user-registration-gcp-cloudrun --destination-run-path=/ --transport-topic=io.automatiko.examples.userRegistration.getuser --location=us-central1
+gcloud eventarc triggers create userregistration --event-filters="type=google.cloud.pubsub.topic.v1.messagePublished" --destination-run-service=user-registration-gcp-cloudrun --destination-run-path=/ --transport-topic=io.automatiko.examples.userRegistration --location=us-central1
+
+gcloud run deploy user-registration-gcp-cloudrun --platform=managed --image=gcr.io/CHANGE_ME/user/user-registration-gcp-cloudrun:1.0.0 --region=us-central1
+````
+
+**NOTE**: `CHANGE_ME` will be replaced with the Google Cloud project id configured in `src/main/resources/application.properties` during the build.
+
+This will provision all required components such as
+
+- PubSub Topics
+- Eventarc Triggers
+- Service deployment
+
+#### Run the service on Google Cloud Run
+
+Once service is deployed you can publish first message (for example using Google Console) to the `io.automatiko.examples.userRegistration` topic with following content.
+
+
+```
+{"user" : {"email" : "mike.strong@email.com",  "firstName" : "mike",  "lastName" : "strong"}}
+```
+
+This will go via number of events being exchanged over the Google Cloud PubSub topics and invoking functions defined in the workflow.
+
+
+## Summing up
+
+Workflow as a function flow implements common scenario for serverless usage where
+individual functions build up a complete business case. It allows to use workflow to design complete business logic and then slice it into functions that are composed into a function flow based on the actual logic defined by the workflow definition. Yet each function can be invoked at anytime making the workflow to act as a blueprint that can start at any place and continue according to defined flow of activities aka functions.
+
+Workflow as a function flow takes advantage of Knative Eventing as the backbone of the communication to enable
+
+- scalability as each function is invoked via Knative broker
+- all data exchange is done with Cloud Events
+- flexibility at which point in the workflow definition instance should be started
+- use "listen to yourself" principle to avoid long running actions impacting overall performance
```

Comment:
```suggestion
Workflow as a function flow takes advantage of Knative Eventing as the backbone of the communication to enable:

- Scalability as each function is invoked by using the Knative broker.
- All data exchange is done with Cloud Events.
- Flexibility with regard to which point in the workflow definition instance should be started.
- Use "listen to yourself" principle to avoid long running actions impacting overall performance.
```